### ISSUE TITLE
twisted.protocols.ftp.FTP does too much in __init__

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5119]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#402|
|Type|defect|
|Created|2003-11-19 06:16:19Z|

Attachments:

* [ftp__init__-patch.txt](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5e6/5e63673cbc075e2d21cef6bde4c591c263380bee/8e10df6f5032f5857901747030645f53aed874b0.txt) (3043 bytes) - added by exarkun on 2003-11-19 06:16:19Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__402 402
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1069222579000000 1069222579000000
changetime__1469597006116450 1469597006116450
version__ 
owner__slyphon slyphon
cc__exarkun cc__slyphon
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
Patch attached to get rid of __init__ completely
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="slyphon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|slyphon commented|
|-|-|

```
#!html
<pre>
applied patch, testing
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="slyphon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|slyphon commented|
|-|-|

```
#!html
<pre>
done!
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub commented|
|-|-|

In changeset bb50a691b0c17bfc9c99e53e1d30def15cf969e4
```
#!CommitTicketReference repository="" revision="bb50a691b0c17bfc9c99e53e1d30def15cf969e4"
Merge pull request #402 from twisted/8665-rodrigc-socks

Author: rodrigc
Reviewer: alex
Fixes: #8665

Port twisted.protocols.socks to Python 3
```
